profile-widget/qmlprofile.cpp

```diff
@@ -85,19 +94,24 @@ int QMLProfile::diveId() const
 	return m_diveId;
 }
 
-void QMLProfile::setDiveId(int diveId)
+void QMLProfile::updateProfile(QString reason)
 {
-	m_diveId = diveId;
-	if (m_diveId < 0)
-		return;
-	struct dive *d = get_dive_by_uniq_id(diveId);
+	struct dive *d = get_dive_by_uniq_id(m_diveId);
 	if (!d)
 		return;
 	if (verbose)
-		qDebug() << "setDiveId(" << d->number << ")";
+		qDebug() << reason << "dive #" << d->number << "x" << QString::number(m_xOffset, 'f', 1) << "y" << QString::number(m_yOffset, 'f', 1);
 	m_profileWidget->plotDive(d, true);
 }
 
+void QMLProfile::setDiveId(int diveId)
+{
+	m_diveId = diveId;
+	if (m_diveId < 0)
+		return;
+	updateProfile(QString("setDiveId"));
```

Comment:
Is the `QString` even needed. I would imagine that `const char *` is automatically cast to `QString` when passed as an argument.

If you want to use the `QString`, use `QStringLiteral` instead for the 1-2 μs gain. :-P

Reply:
The string was useful when I started debugging the code... and frankly, if we ever run into a situation where we get runaway update requests, we might be grateful for it. But I can of course optimize this with `QStringLiterals`

Reply:
actually, now that there are only two calls to `updateProfile()` left, there really isn't a point anymore. So I removed the whole reason argument.

Reply:
I think you misunderstood my point:
Couldn't you simply write
```
        updateProfile("setDiveId");
```
I would presume that since `QString` has a conversion constructor from `const char *`, this should work. It could of course also lead to an error owing to ambiguity.

Reply:
I understood your point. I just decided that dropping that 'reason' argument was the more appropriate thing to do. Initially all of the setters called `updateProfile()` and the debug output got really confusing. That's why I added the `reason` argument in the first place.
But now that I don't do that anymore, it's not needed. So why clutter the code and waste critical µs...